internal/vault_reader.go

```diff
@@ -0,0 +1,293 @@
+	createSecretStoreClientOnce.Do(InitializeClient(vaultSpec))
+	return SecretClient, SecretClientCreationError
+}
+
+func InitializeClient(vaultSpec *rabbitmqv1beta1.VaultSpec) func() {
+	return func() {
+		// VAULT_ADDR environment variable will be the address that pod uses to communicate with Vault.
+		config := vault.DefaultConfig() // modify for more granular configuration
+		vaultClient, err := vault.NewClient(config)
+		if err != nil {
+			SecretClientCreationError = fmt.Errorf("unable to initialize Vault client: %w", err)
+			return
+		}
+
+		go renewToken(vaultClient, vaultSpec, FirstLoginAttemptResultCh)
+		err = <-FirstLoginAttemptResultCh
+		if err != nil {
+			SecretClientCreationError = fmt.Errorf("unable to login to Vault: %w", err)
+			return
+		}
+
+		SecretClient = VaultClient{
+			Reader: &VaultSecretReader{client: vaultClient},
+		}
+	}
+}
+
+func (vc VaultClient) ReadCredentials(path string) (CredentialsProvider, error) {
+	secret, err := vc.Reader.ReadSecret(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read Vault secret: %w", err)
+	}
+
+	if secret == nil {
+		return nil, errors.New("returned Vault secret is nil")
+	}
+
+	if secret != nil && secret.Warnings != nil && len(secret.Warnings) > 0 {
+		return nil, fmt.Errorf("warnings were returned from Vault: %v", secret.Warnings)
+	}
+
+	if secret.Data == nil {
+		return nil, errors.New("returned Vault secret has a nil Data map")
+	}
+
+	if len(secret.Data) == 0 {
+		return nil, errors.New("returned Vault secret has an empty Data map")
+	}
+
+	if secret.Data["data"] == nil {
+		return nil, fmt.Errorf("returned Vault secret has a Data map that contains no value for key 'data'. Available keys are: %v", availableKeys(secret.Data))
+	}
+
+	data, ok := secret.Data["data"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("data type assertion failed for Vault secret of type: %T and value %#v read from path %s", secret.Data["data"], secret.Data["data"], path)
+	}
+
+	username, err := getValue("username", data)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get username from Vault secret: %w", err)
+	}
+
+	password, err := getValue("password", data)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get password from Vault secret: %w", err)
+	}
+
+	return ClusterCredentials{
+		data: map[string][]byte{
+			"username": []byte(username),
+			"password": []byte(password),
+		},
+	}, nil
+}
+
+func getValue(key string, data map[string]interface{}) (string, error) {
+	result, ok := data[key].(string)
+	if !ok {
+		return "", fmt.Errorf("expected %s to be a string but is a %T", key, data[key])
+	}
+
+	return result, nil
+}
+
+func availableKeys(m map[string]interface{}) []string {
+	result := make([]string, len(m))
+	i := 0
+	for k := range m {
+		result[i] = k
+		i++
+	}
+	return result
+}
+
+func login(vaultClient *vault.Client, vaultSpec *rabbitmqv1beta1.VaultSpec) (*vault.Secret, error) {
```

Comment:
@MarcialRosales @georgeharley 
Interesting nuance....all a part of the journey.  If I may try to reiterate the problem and make a suggestion.  Apologies in advance for the over explanation, just confirming my understanding.

1. We have two steps in play for Vault integration.  Vault Auth vs Vault credential access.
2. Vault Auth, utilizing the k8s backend, uses the following components:
     -  Vault Address:  Established via topology operator environment variable.
     -  SA Token:  Currently pulled from the topology-operator container.
     -  Auth Role:  Currently defaults to `messaging-topology-operator`.
     -  Auth Path:  Currently pulled from rabbit cluster CR.
     -  Vault Namespace:  Currently pulled from rabbit cluster CR.
3.  Once auth'ed, the client will request and be granted access to credentials (path sourced from the CR) based on policies setup on the Auth Role.
4. The Vault Namespace and Auth Path on the CR is related to the Auth Role & SA used for the Rabbit Cluster rather than the Topology Operator.

#### MVP Suggestion:
It would seem as if we're close to a configuration where the Topology Operator would be able to cache the Vault token.  The changes needed would be to transition the Auth Path & Vault Namespace components from being CR sourced to being specific to the Auth Role, i.e.(env's of OPERATOR_AUTH_PATH, VAULT_NAMESPACE).

#### MVP Assumptions:
*  Same Vault endpoint is used by all clusters.
*  The same Vault Namespace, or no namespace, is used by all clusters.
*  The Topology Operator role will have access to all of the credential paths.

#### Future Considerations
Not sure if this would be appropriate, however...

Have the Topology Operator leverage all of the components of the Rabbit Cluster to login to Vault.  This would require the Topology Operator to pull the token from the SA secret and use it along with the components in the CR.  This would be more of an "assume role" type of pattern and remove the need to manage the Topology Operator Role.  This would get a bit more involved as we may then find the need to cache the Vault Token per Rabbit cluster?

Reply:
@twhite0 , I appreciate a lot your clear explanation ! 

Your MVP suggestion and assumptions makes a lot of sense, specially for the MVP. The change we have to make is minimal. As you summarized it, we only need to source the auth path and vault namespace from an Environment variable along with the role's environment variable. It makes a lot of sense and I think it is more correct. 

Using a dedicated role+auth_path+namespace settings for the topology operator has advantages and disadvantages. 
The *advantage* is that you have a separate vault access control policy for the cluster operator and for the messaging topology operator. The disadvantage is the extra administrative burden in Vault. 

Your *future considerations* is really what I had in mind from the beginning. It seemed very logical. A vault administrator sets the appropriate role, auth-path, namespace for a cluster and this configuration is used by both, cluster and topology operators. 
The disadvantage is that the access control is **all-or-nothing** per cluster, i.e. either both operators can operate on a cluster or they cannot. And, as you pointed out, the messaging topology operator would have to cache a vault token per cluster. The topology operator would have to use a bounded cache with `max-size` and `TTL` capabilities to limit the amount of tokens to cache. 

If both strategies made sense, we could configure the topology operator with additional setting, vault_auth_strategy with 2 values : **assume_role** where the topology operator assumes all the auth settings from the cluster, or **dedicated_role** where the topology operator uses its own dedicated auth settings.